app/controllers/user_preferences_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UserPreferencesController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :check_acting_as
+
+  def index
+  end
+
+  def edit
+    @user_preference = UserPreference.find params[:id]
+  end
+
+  def update
+    @user_preference = UserPreference.find params[:id]
```

Comment:
We should probably limit this to `current_user.user_preferences.find`. That way we don't have to worry about abilities too much.